DNS/Server/DnsServer.cs

```diff
@@ -61,6 +61,18 @@ public class DnsServer: IDisposable {
             if (run) {
                 try {
                     udp = new UdpClient(ip);
+
+                    // Configure UdpClient to ignore PORT_UNREACHABLE ICMP messages.
+                    //
+                    // This safeguards our server from ICMP-triggered exceptions in a scenario
+                    // where requester unbinds its socket prior to receiving our response.
+                    const int SIO_UDP_CONNRESET = unchecked((int)0x9800000C);
+                    // In-value: false (disable PORT_UNREACHABLE reporting)
+                    var inValue  = new byte[] { 0, 0, 0, 0 };
+                    var outValue = new byte[] { 0, 0, 0, 0 };
+
+                    udp.Client.IOControl(SIO_UDP_CONNRESET, inValue, outValue);
```

Comment:
On my mac this line throws a `System.PlatformNotSupportedException` exception with the message `Socket.IOControl handles Windows-specific control codes and is not supported on this platform`. The same seems to happening on travis.

Maybe a better solution is to wrap the `BeginReceive` calls in a `try/catch` and handle the `SocketException` by opening a new `UdpClient`. But that requires a bit more refactoring. I'll think about the best approach.

Reply:
I've ran into the same issue in case the client closed the socket already. Then this line here runs into a SocketException.
`if (run) udp.BeginReceive(receiveCallback, null);`

Reply:
Use `RuntimeInformation.IsOSPlatform` for detect windows